### ISSUE TITLE
Error while using transactions: attempt to initiate a new Adaptive Server operation with results pending

### ISSUE THREAD

**Issue by xpohoc69** [yiisoft/yii2#19036]
I use mssql, version 14.0.3223.3. My config ```'dsn' => 'dblib:version=7.2;host=10.199.??.???;dbname=web_trunk;charset=utf8'```, ```'driverName' => 'mssql'```

I updated Yii from 2.0.36 to 2.0.43 version and got next error in my code. ```SQLSTATE[HY000]: General error: 20019 Attempt to initiate a new Adaptive Server operation with results pending [20019] (severity 7) []``` (https://monosnap.com/file/sTRWKER32VCSbfW3Qb1lK2N5NWRgjZ). I use a transaction. Inside transaction I call a procedure and use save() method of ORM. The error occures in ORM. I found out that error happens due to #18214, #18287 changes. If I return old mssql\QueryBuilder::insert() method (https://monosnap.com/file/Ezs0BvzIBciNgglsT7KjFOL9nlvzD2), from version 2.0.37 -- it works fine.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.43
| PHP version      | 7.4.13
| Operating system | Windows 10, docker container, php:7.4-fpm

**Comment by xpohoc69** [comment]
@darkdef @samdark would you please take a look?

**Comment by samdark** [comment]
What is your MSSQL version?

**Comment by xpohoc69** [comment]
@samdark 14.0.3223.3

**Comment by xpohoc69** [comment]
@samdark found another issue with mssql #19040

**Comment by xpohoc69** [comment]
I conducted such tests

Attempted valid request:

      Yii :: $ app-> web-> transaction (function () {
        $ model = new IframeAnalytics ();
        $ model-> iframe_key = 'bjUzqWBMIYZVJLiDPjs2Ii1Pdb1HSdxB';
        $ model-> save ();
      });

1.dblib + driverName: dblib - falls with 502 on commit
2.dblib + driverName: mssql - inTransaction () does not work, always returns false, accordingly you cannot commit()
3.odbc + driverName: mssql - inTransaction () does not work, it always returns false, accordingly you cannot commit()
4. odbc + driverName: dblib - everything works, no problem

Inherited from ```yii\db\mssql\PDO```. I registered ```inTransaction() {return true;}``` in it in an attempt to fix items 2-3 above. I connected the config ```'pdoClass' => app\components\FixTrPDO::class```

1.dblib + driverName: dblib - SQLSTATE [HY000]: General error: 20019 Attempt to initiate a new Adaptive Server operation with results pending [20019] (severity 7) []
2.dblib + driverName: mssql - SQLSTATE [HY000]: General error: 20019 Attempt to initiate a new Adaptive Server operation with results pending [20019] (severity 7) []
3. odbc + driverName: mssql - everything works, no problem
4. odbc + driverName: dblib - everything works, no problem
